crates/sui-graphql-rpc/src/server/data_provider.rs

```diff
@@ -44,7 +47,60 @@ pub(crate) async fn fetch_obj(
             val.data.unwrap()
         }
     };
-    Ok(Some(convert_obj(g)))
+    Ok(Some(convert_obj(&g)))
+}
+
+pub(crate) async fn fetch_owned_objs(
+    cl: &SuiClient,
+    owner: &SuiAddress,
+    first: Option<u64>,
+    after: Option<String>,
+    last: Option<u64>,
+    before: Option<String>,
+    _filter: Option<ObjectFilter>,
+) -> Result<Connection<String, Object>> {
+    if before.is_some() && after.is_some() {
+        return Err(Error::new("before and after must not be used together"));
+    }
+    if first.is_some() && last.is_some() {
+        return Err(Error::new("first and last must not be used together"));
+    }
+    if before.is_some() || last.is_some() {
+        return Err(Error::new("reverse pagination is not supported"));
+    }
+
+    let count = first.map(|q| q as usize);
+    let native_owner = NativeSuiAddress::from(owner);
+    let query = SuiObjectResponseQuery::new_with_options(SuiObjectDataOptions::full_content());
+
+    let cursor = match after {
+        Some(q) => Some(
+            NaiveObjectID::from_hex_literal(&q)
+                .map_err(|w| Error::new(format!("invalid object id: {}", w)))?,
+        ),
+        None => None,
+    };
+
+    let pg = cl
+        .read_api()
+        .get_owned_objects(native_owner, Some(query), cursor, count)
+        .await?;
+
+    pg.data.iter().try_for_each(|n| {
```

Comment:
Do we need to return an error if not every owned object is successfully fetched? I think this would be a breaking change from rpc 1.0

Reply:
We can mark this as a future improvement after discussing with team. Right now the connection story in gql doesnt allow for errors within the paginated values. We will probably need a special type which can embed the enum.

>  I think this would be a breaking change from rpc 1.0

This isn't necessarily a bad thing

Reply:
Yah that's fair

Reply:
@oxade, are we recording these future improvements somewhere?  I know we've talked about partial errors as well in the past + if it's something that we already do in the current RPC implementation, I want to make sure that even if we don't do it as part of the MVP, we have a place we can go back to see them, because this is something I'd probably want us to re-evaluate before we go live.

Feel free to re-resolve if there's already such a place, I just want to make sure this context isn't lost.